spec.emu

```diff
@@ -1030,7 +1036,7 @@ contributors: Nicolò Ribaudo
               1. Append _module_ to _seen_.
               1. If _module_ is not a Cyclic Module Record, return ~unused~.
               1. If _module_.[[Status]] is either ~evaluating~ or ~evaluated~, return ~unused~.
-              1. If _module_.[[HasTLA]] is *true*, then
+              1. If _module_.[[HasTLA]] is *true* or _module_.[[Status]] is ~evaluating-async~, then
```

Comment:
(continuing from https://github.com/tc39/proposal-defer-import-eval/pull/36#issuecomment-2030804500)

I still don't understand how this can work.

Consider this example:
```js
// a
import "b";
throw new Error();
```
```js
// b
await sleep(1000);
```
```js
// entrypoint
const p1 = import("a");
await sleep(200);
const p2 = import.defer("a");

console.log(await Promise.allSettled([p1, p2]));
```

The expected result here is that `p1` is rejected, and `p2` is fulfilled (because `p2` only needs to wait for the async dependencies, i.e. `b`). However, with this change `p2` is also rejected because `import("a")` happened before `import.defer("a")`, so it puts `a` in `~evaluating-async~` state and affects the behavior of `import.defer("a")`.

Reply:
Great point and discussion today on this, let's move the discussion to https://github.com/tc39/proposal-defer-import-eval/issues/41.

I've reverted this change for now, to rather follow-up separately.